stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -588,8 +588,12 @@ internal class ChannelLogic(
                     is NotificationChannelTruncatedEvent -> removeMessagesBefore(event.createdAt)
                     is TypingStopEvent -> channelStateLogic.setTyping(event.user.id, null)
                     is TypingStartEvent -> channelStateLogic.setTyping(event.user.id, event)
-                    is MessageReadEvent -> channelStateLogic.updateRead(event.toChannelUserRead())
-                    is NotificationMarkReadEvent -> channelStateLogic.updateRead(event.toChannelUserRead())
+                    is MessageReadEvent -> if (event.thread == null) {
+                        channelStateLogic.updateRead(event.toChannelUserRead())
+                    }
```

Comment:
What about thread?
aren't we updating it?
Or it is already updated somewhere else?

Reply:
I believe the thread is handled in `QueryThreadsLogic` 🤔